File: backend/master/internal/model/log.go
package model

import "time"

// OperationLog 记录管理员操作。
type OperationLog struct {
	ID         uint      `gorm:"primaryKey" json:"id"`
	AdminID    uint      `gorm:"index" json:"admin_id"`
	Action     string    `gorm:"size:100;not null" json:"action"`
	TargetType string    `gorm:"size:64" json:"target_type"`
	TargetID   *uint     `json:"target_id"`
	Details    string    `gorm:"type:text" json:"details"`
	IPAddress  string    `gorm:"size:64" json:"ip_address"`
	CreatedAt  time.Time `json:"created_at"`

	Admin Admin `json:"admin,omitempty"`
}

File: frontend/admin/src/api/log.ts
import request from '@/utils/request'

// Operation Log Interface
export interface OperationLog {
    id: number
    admin_id: number
    admin_name: string
    action: string
    target_type: string
    target_id?: number
    description: string
    ip_address: string
    user_agent?: string
    request_body?: string
    response_body?: string
    created_at: string
}

// Log Filter
export interface LogFilter {
    admin_id?: number
    action?: string
    target_type?: string
    page?: number
    page_size?: number
}

// Get Operation Log List
export function getOperationLogs(params: LogFilter) {
    return request<{
        items: OperationLog[]
        total: number
        page: number
        page_size: number
    }>({
        url: '/admin/logs',
        method: 'get',
        params
    })
}
